src/tools/wp/bir_wpLib.sml

```diff
@@ -255,25 +258,24 @@ struct
               )
       val thm = SPECL [wps, var_wps1] prog_l_thm
 
-      (* this took a while, it should not have been?! *)
+      (* FIXME: This seems to take some time, is that normal? *)
       val thm = MP thm wps_bool_sound_thm
 
-      (* this takes a bit, not anymore? *)
-      val wps_eval_restrict_consts =
-        !bir_wp_comp_wps_iter_step2_consts;
+      (* FIXME: This seems to take some time, is that normal? *)
+      val wps_eval_restrict_consts = !bir_wp_comp_wps_iter_step2_consts;
+      val obs_ty = (hd o snd o dest_type o type_of) program
       val wps1_thm =
         computeLib.RESTR_EVAL_CONV wps_eval_restrict_consts
           (list_mk_comb
             (* TODO: Add to bir_wpSyntax *)
-            (``bir_wp_exec_of_block:'a bir_program_t ->
+            (inst [Type `:'obs_ty` |-> obs_ty]
```

Comment:
I cannot test this right now. Could you explain to me briefly why this makes sense?

Reply:
If the observation type in `program` isn't `'a`, then `list_mk_comb` will fail because of type mismatch. So I'm instantiating the observation type to the program's obs type.

Reply:
oh i see, that makes a lot of sense